src/module/actor/character/sheet.ts

```diff
@@ -572,6 +574,155 @@ class CharacterSheetPF2e extends CreatureSheetPF2e<CharacterPF2e> {
         }
 
         // CRAFTING
+        const craftingTab = htmlQuery(html, ".tab.crafting");
+
+        for (const element of htmlQueryAll<HTMLLIElement>(craftingTab, "li.formula-item")) {
+            const quantity = htmlQuery<HTMLInputElement>(element, "input[data-action=enter-quantity]");
+            quantity?.addEventListener("change", async () => {
+                const itemUUID = element.dataset.itemId ?? "";
+                const formula = this.knownFormulas[itemUUID];
+                const minBatchSize = formula.minimumBatchSize;
+                const newValue = Number(quantity.value) || minBatchSize;
+                if (newValue < 1) return;
+
+                const entrySelector = htmlClosest(element, "li.crafting-entry")?.dataset.entrySelector;
+                if (entrySelector) {
+                    const craftingEntry = await this.actor.getCraftingEntry(entrySelector);
+                    if (!craftingEntry) throw ErrorPF2e("Crafting entry not found");
+
+                    const index = element.dataset.itemIndex;
+                    await craftingEntry.setFormulaQuantity(Number(index), itemUUID ?? "", newValue);
+                    return;
+                }
+                this.formulaQuantities[formula.uuid] = Math.max(newValue, minBatchSize);
+                this.render();
+            });
+            for (const button of htmlQueryAll(
+                element,
+                "[data-action=increase-quantity], [data-action=decrease-quantity]"
+            )) {
+                button.addEventListener("click", async () => {
+                    if (!quantity) return;
+                    const itemUUID = element.dataset.itemId ?? "";
+                    const formula = this.knownFormulas[itemUUID];
+                    const minBatchSize = formula.minimumBatchSize;
+                    const step = button.dataset.action === "increase-quantity" ? minBatchSize : -minBatchSize;
+                    const newValue = (Number(quantity.value) || step) + step;
+                    if (newValue < 1) return;
+
+                    const entrySelector = htmlClosest(element, "li.crafting-entry")?.dataset.entrySelector;
+                    if (entrySelector) {
+                        const craftingEntry = await this.actor.getCraftingEntry(entrySelector);
+                        if (!craftingEntry) throw ErrorPF2e("Crafting entry not found");
+
+                        const index = element.dataset.itemIndex;
+                        await craftingEntry.setFormulaQuantity(Number(index), itemUUID ?? "", newValue);
+                        return;
+                    }
+                    this.formulaQuantities[formula.uuid] = Math.max(newValue, minBatchSize);
+                    this.render();
+                });
+            }
+
+            const craftButton = htmlQuery(element, "a[data-action=craft-item]");
```

Comment:
Was this a single button or one for each formula?

Reply:
That's the 🔨 button on each formula.